orangecontrib/text/corpus_to_network.py

```diff
@@ -0,0 +1,292 @@
+        ------
+        ValueError
+            If threshold <= 0 or window_size <= 0 or freq_threshold <= 0
+            or if document_nodes is not instance of bool.
+        """
+
+        if threshold <= 0:
+            raise ValueError("Threshold must be at least 1.")
+        if window_size <= 0:
+            raise ValueError("Window size must be at least 1.")
+        if freq_threshold <= 0:
+            raise ValueError("Frequency threshold must be at least 1.")
+        if not isinstance(document_nodes, bool):
+            raise ValueError("Document_nodes must be bool.")
+        
+        self.state = state
+        self._cache_params() # remember old parameters
+        try:
+            self.last_called_nodes = document_nodes
+            if (document_nodes and
+                    threshold == self.document_threshold and
+                    self.document_network is not None):
+                # same network already exists
+                if self.state is not None:
+                    state.set_progress_value(100.0)
+                return self.document_network
+            if (not document_nodes and
+                    threshold == self.word_threshold and
+                    window_size == self.window_size and
+                    freq_threshold == self.freq_threshold and
+                    self.word_network is not None):
+                # same network already exists
+                if self.state is not None:
+                    state.set_progress_value(100.0)
+                return self.word_network
+            if document_nodes:
+                self.document_threshold = threshold
+                if self.document_matrix is None:
+                    self._compute_document_matrix() # generate adjacency matrix
+                self._generate_document_network() # construct network
+                return self.document_network
+            else:
+                if (window_size != self.window_size or
+                        self.word_matrix is None or
+                        freq_threshold != self.freq_threshold):
+                    if self.word2ind is None:
+                        self._compute_word2ind()
+                    self.window_size = window_size
+                    self.freq_threshold = freq_threshold
+                    self._compute_word_matrix() # generate adjacency matrix
+                self.word_threshold = threshold
+                self._generate_word_network() # construct network
+                return self.word_network
+        except Exception:
+            self._restore_params()
+
+
+    def _compute_document_matrix(self):
+        self.document_matrix = np.zeros((len(self.corpus), len(self.corpus)))
+        num_ticks = (len(self.corpus) * (len(self.corpus) - 1)) // 2
+        ticks = iter(np.linspace(0., 90., num_ticks))
+        for i in range(len(self.corpus)):
+            for j in range(i + 1, len(self.corpus)):
+                w = set(self.corpus.tokens[i]).intersection(
```

Comment:
Is it ok that you use tokens here or should we use `ngrams`? `tokens` include preprocessed corpus without Ingram preprocessing step, while `ngrams` includes all preprocessing.